src/v/test_utils/scoped_config.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "config/base_property.h"
+#include "config/configuration.h"
+
+// Scoped wrapper around config::shard_local_cfg() that tracks get() calls and,
+// upon destructing, resets any properties that were potentially mutated.
+class scoped_config {
+public:
+    ~scoped_config() {
+        for (auto& p : _properties_to_reset) {
+            config::shard_local_cfg().get(p).reset();
```

Comment:
I think it has to do this on all shards. Something like:
```
        ss::smp::invoke_on_all([&] {
            config::shard_local_cfg().get(p).reset();
        }).ge();
```

and same is for setting the value. We have tests which are using more than one shard and setting config parameters (without resetting).  For instance `throughput_limits_fixture` in the `produce_consume_test.cc`

Reply:
Done

Reply:
Actually the setting part of this seems a bit more involved. I'm not sure it's the right thing to do to always do it on every core. I'm leaning towards forcing test authors to manage this per core with a sharded service or somesuch.